### PR TITLE
Implement dynamic list of regions.

### PR DESCRIPTION
**Pull request by golenkovm**
This is a fix for #146

### CODE REVIEW

**Comment by golenkovm** [comment]
The section now looks like
image

A new `catch` clause is need to cover this exception:
image

File: classes/local/store/s3/client.php [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
 use tool_objectfs\local\manager;
 use tool_objectfs\local\store\object_client_base;
+use local_aws\admin_settings_aws_region;
```
Comment (brendanheywood) [inline classes/local/store/s3/client.php:32]: will this explode if not found?

File: lang/en/tool_objectfs.php [hunk 2]
```diff
@@ -93,6 +93,7 @@
 $string['settings:aws:region_help'] = 'Amazon S3 API gateway region.';
 $string['settings:aws:base_url'] = 'Base URL';
 $string['settings:aws:base_url_help'] = 'Alternate url for cnames or s3 compatible endpoints. Leave blank for normal S3 use.';
+$string['settings:aws:upgradeneeded'] = 'Please upgrade \'local_aws\' plugin up to 2020051200 version.';
```
Comment (brendanheywood) [inline lang/en/tool_objectfs.php:96]: you should say, "at least". Probably better to say 'to the latest supported version'

File: classes/local/store/s3/client.php (hunk 1 above)
Comment (golenkovm) [reply classes/local/store/s3/client.php:32]: Tested this. It's not exploding.

File: lang/en/tool_objectfs.php (hunk 2 above)
Comment (golenkovm) [reply lang/en/tool_objectfs.php]: Set to `'Please upgrade \'local_aws\' plugin to the latest supported version.'`

**Pull request merged**